Add a test case for issue #46 / pr #48  for Windows

AirQuick opened the pull request:
This change makes the tests on AppVeyor parity with Travis

* Mimic `=~` operator of Bats
* Copy [the test case](https://github.com/xspec/xspec/blob/ba5f97c903d0022d322d7af7e379b6fbcaf35ee6/test/xspec.bats#L172-L177)

cirulls commented on test/xspec-bat.cmd:
```diff
@@ -258,7 +258,17 @@ setlocal
+
+    call :run ..\bin\xspec.bat ..\test\xspec-46.xspec
+    call :verify_retval 0
+    call :verify_line 4 r "Testing with"
```
This is fine now but I think it may break after #65 is merged into `master` (see notes in #65). I think this may need to be changed to:
```winbatch
call :verify_line 5 r "Testing with"
```

AirQuick replied:
Good point. Let's handle this after #65.

AirQuick commented:
Updated to keep up with the latest changes.

**test/xspec-bat.cmd**
* Incorporated https://github.com/xspec/xspec/commit/82a3d490a5d2a8e8349fee81c3eb5c9ee3eb298a#diff-a9965dc9cb28e7c25ee137ad862cbe38

**test/run-xspec-tests.cmd** (This wasn't the scope of the original pull request.)
* It used to avoid using `test/xspec` dir. Now it uses the same output directory and log file as the shell script.
